app/common/state/tabContentState.js

```diff
@@ -152,7 +152,8 @@ const tabContentState = {
       // If closeIcon is fixed then there's no room for another icon
       !tabContentState.hasFixedCloseIcon(state, frameKey) &&
       // completely hide it for small sizes
-      !hasBreakpoint(frame.get('breakpoint'), ['mediumSmall', 'small', 'extraSmall', 'smallest'])
+      !hasBreakpoint(frame.get('breakpoint'),
+        ['medium', 'mediumSmall', 'small', 'extraSmall', 'smallest'])
```

Comment:
the feature works as expected but turns out to have bad UI. It wasn't introduced in your PR.